src/schema/posts.ts

```diff
@@ -1087,7 +1087,25 @@ export const resolvers: IResolvers<any, Context> = {
         partialPost.private ||
         sourceTypesWithMembers.includes(postSource.type)
       ) {
-        await ensureSourcePermissions(ctx, partialPost.sourceId);
+        try {
+          await ensureSourcePermissions(ctx, partialPost.sourceId);
+        } catch (permissionError) {
+          if (permissionError instanceof ForbiddenError) {
+            const forbiddenError = permissionError as ForbiddenError;
+
+            const forbiddenErrorForPost = new ForbiddenError(
+              permissionError.message,
+              {
+                ...forbiddenError.extensions,
+                postId: partialPost.id,
+              },
+            );
```

Comment:
This can be bundled in new Error class eg `PostForbiddenError` but I don't think it is needed at this point.